frontend/tests/util/blocks.ts

```diff
@@ -0,0 +1,238 @@
+                await expect(page.getByLabel("Oldest first")).toBeChecked();
+                await saveButton.click();
+
+                await expect(
+                    page.getByRole("button", { name: "Choose video order" })
+                ).toHaveText("Oldest first");
+            });
+
+            await test.step("Layout options can be changed", async () => {
+                await test.step("Title and videos (initial)", async () => {
+                    await expect(
+                        page.getByRole("heading", { name: newSeries.title })
+                    ).toBeVisible();
+                });
+
+                await test.step("Videos only", async () => {
+                    await editButton.click();
+                    await page.getByLabel("Show title").setChecked(false);
+                    await page.getByLabel("Show description").setChecked(false);
+                    await saveButton.click();
+
+                    await expect(
+                        page.getByRole("heading", { name: newSeries.title })
+                    ).not.toBeVisible();
+                    await expect(
+                        page.getByText("Cat software running on dog hardware.")
+                    ).not.toBeVisible();
+                });
+
+                await test.step("Description and videos", async () => {
+                    await editButton.click();
+                    await page.getByLabel("Show description").setChecked(true);
+                    await saveButton.click();
+
+                    await expect(
+                        page.getByRole("heading", { name: newSeries.description })
+                    ).not.toBeVisible();
+                    await expect(page.getByText(newSeries.description).first()).toBeVisible();
+                });
+
+                await test.step("Title, description and videos", async () => {
+                    await editButton.click();
+                    await page.getByLabel("Show title").setChecked(true);
+                    await page.getByLabel("Show description").setChecked(true);
+                    await saveButton.click();
+
+                    await expect(
+                        page.getByRole("heading", { name: newSeries.title })
+                    ).toBeVisible();
+                    await expect(page.getByText(newSeries.description).first()).toBeVisible();
+                });
+            });
+            break;
+        }
+        case "video": {
+            await test.step("Video can be changed", async () => {
+                const input = page.getByRole("combobox");
+                await input.pressSequentially("Dual Stream Cats");
+                await page.getByRole("option", { name: "Dual Stream Cats"}).click();
+
+                await page.keyboard.press("Enter");
+
+                await expect(page
+                    .getByRole("heading", { name: "Video Of A Tabby Cat" })
+                    .first())
+                    .toBeVisible();
```

Comment:
Again, `first()` +  `toBeVisible`. Now that I think about it again, even without a `not`, this is fishy. Because again, `first()` implies there is some random other node that contains this text. And while that's present this check is not really checking anything. 

Maybe something like `.toHaveCount(2)` is rather something you want? But I'm just guessing here.